modules/shared/src/main/scala/org/tessellation/scalacheck/generators/trust.scala

```diff
@@ -0,0 +1,64 @@
+package org.tessellation.scalacheck.generators
+
+import org.tessellation.schema.TrustValueRefinement.TrustValueRefinement
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.schema.trust._
+import org.tessellation.security.hex.Hex
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.refineV
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.{Arbitrary, Gen}
+
+object trust {
+
+  implicit val arbitraryHexString: Arbitrary[String] = Arbitrary {
+    Gen.choose(0, 0xffffff).flatMap(_.toHexString)
+  }
+
+  implicit val arbitraryTrustDouble: Arbitrary[Double] = Arbitrary {
+    Gen.choose(-1.0, 1.0)
+  }
+
+  implicit val arbitraryOptionalTrustValue: Arbitrary[Option[Double]] = Arbitrary {
+    Gen.option(Gen.choose(-1.0, 1.0))
+  }
+
+  val genHex = arbitrary[String].flatMap(Hex(_))
+
+  val genScore = arbitrary[Double]
+    .flatMap(refineV[TrustValueRefinement](_))
+    .flatMap(_ match {
+      case Left(_)      => Option[Score](Score(0.0))
+      case Right(value) => Option[Score](Score(value))
+    })
```

Comment:
instead of pattern match, you can use
```scala
.flatMap(_.toOption.getOrElse(Score(0.0))
```
or even, add `empty` property to the Score (see: Hash or TransactionReference)
and then:
```scala
.flatMap(_.toOption.getOrElse(Score.empty))
```

Reply:
The generators have been simplified.